packages/nx/src/command-line/init/init-v2.ts

```diff
@@ -0,0 +1,176 @@
+import { existsSync } from 'fs';
+import { PackageJson } from '../../utils/package-json';
+import { prerelease } from 'semver';
+import { output } from '../../utils/output';
+import { getPackageManagerCommand } from '../../utils/package-manager';
+import { generateDotNxSetup } from './implementation/dot-nx/add-nx-scripts';
+import { runNxSync } from '../../utils/child-process';
+import { readJsonFile } from '../../utils/fileutils';
+import { nxVersion } from '../../utils/versions';
+import {
+  addDepsToPackageJson,
+  askAboutNxCloud,
+  createNxJsonFile,
+  runInstall,
+  updateGitIgnore,
+} from './implementation/utils';
+import { prompt } from 'enquirer';
+import { execSync } from 'child_process';
+import { globSync } from '../../native';
+import { addNxToAngularCliRepo } from './implementation/angular';
+
+export interface InitArgs {
+  interactive: boolean;
+  nxCloud?: boolean;
+  useDotNxInstallation?: boolean;
+  integrated?: boolean; // For Angular projects only
+}
+
+export async function initHandler(options: InitArgs) {
+  const version =
+    process.env.NX_VERSION ?? (prerelease(nxVersion) ? 'next' : 'latest');
+  if (process.env.NX_VERSION) {
+    output.log({ title: `Using version ${process.env.NX_VERSION}` });
+  }
+  if (existsSync('package.json') && !options.useDotNxInstallation) {
+    const repoRoot = process.cwd();
+    const cacheableOperations: string[] = [];
+    createNxJsonFile(repoRoot, [], cacheableOperations, {});
+
+    const pmc = getPackageManagerCommand();
+
+    updateGitIgnore(repoRoot);
+
+    // TODO(jack): Remove this Angular logic once `@nx/plugin` is compatible with PCv3.
+    let plugins: string[];
+    if (existsSync('angular.json')) {
+      plugins = [];
+      await addNxToAngularCliRepo({
+        ...options,
+        integrated: !!options.integrated,
+      });
```

Comment:
We should do this as early as possible, in line 36, and we also need to exit after this. Otherwise, we'll duplicate some operations. The `addNxToAngularCliRepo` function already includes all the steps performed here unrelated to the new logic adding plugins (e.g. `createNxJsonFile`, `updateGitIgnore`, setting up Nx Cloud, etc.).

Reply:
Moved the `.nx` and Angular logic to the top with early return.